src/genai/prompt_pattern.py

```diff
@@ -75,6 +81,37 @@ def from_str(cls, prompt: str):
         pt._create(url=prompt, literal=True)
         return pt
 
+    @classmethod
+    def from_watsonx(cls, credentials: Credentials, name: str = None, template: dict = None, id: str = None):
+        # Cases :
+        # fetching an existing template : name OR id
+        # updating an existing template : template + (name OR id)
+        # creating a new template       : template + name
+
+        # Is there an existing template with the name or id provided?
+        try:
+            saved_template = PromptServiceInterface.load_template(credentials=credentials, id=id, name=name)
```

Comment:
IMO this exception should be specialized to distinguish between other exceptions and the case when such a template does not exist. Otherwise sometimes there could be other exceptions from `load_templates` -- what this piece of code here would do is define None in such a case.

If we are not specializing the exception here, then in that case `load_template` should handle the exception and return `None` if no such template found.